File: Kfz/src/kfzwerkstatt/Verkaeufer.java
package kfzwerkstatt;

import java.util.GregorianCalendar;

public class Verkaeufer {

	public enum Prioritaet {
		DRINGEND, WICHTIG, NORMAL;
	}

	private int lohnProZeiteinheit = 3;

	private final String vName;
	private final String vVorname;
	private final String vId;

	public float sumVerkaeufe;

	public Verkaeufer(String name, String vorname, String id) {
		this.vId = id;
		this.vName = name;
		this.vVorname = vorname;
	}

	public void verkaufeDienstleitung(Kunde kunde, Prioritaet prio,
			String fehlerbeschreibung, Mechaniker m) {

		long startzeit = GregorianCalendar.getInstance().getTimeInMillis();
		synchronized (this) {
			try {
				// Hier wird das Arbeiten simuliert
				this.wait((int) (Math.random() * 100));
			} catch (InterruptedException e) {
			}
		}

		long endzeit = GregorianCalendar.getInstance().getTimeInMillis();
		int dauer = (int) (endzeit - startzeit);

		// Der Verkäufer berechnet auch Kosten für seine Dienstleistung :-D
		// ...quasi Trinkgeld :-D
		int lohn = dauer * lohnProZeiteinheit;
		sumVerkaeufe += lohn;
		kunde.addBezahlt(lohn);

		m.bearbeiteJob(kunde, prio, fehlerbeschreibung);

	}

	public String getVName() {
		return vName;
	}

	public String getVVorname() {
		return vVorname;
	}

	public String getVId() {
		return vId;
	}

	/**
	 * @return the lohnProZeiteinheit
	 */
	public int getLohnProZeiteinheit() {
		return lohnProZeiteinheit;
	}

	/**
	 * @param lohnProZeiteinheit
	 *            the lohnProZeiteinheit to set
	 */
	public void setLohnProZeiteinheit(int lohnProZeiteinheit) {
		this.lohnProZeiteinheit = lohnProZeiteinheit;
	}

}

File: KfzScala/src/kfzwerkstatt/Verkaeufer.scala
package kfzwerkstatt

object Verkaeufer {

  object Prioritaet extends Enumeration {
    type Prioritaet = Value
    val DRINGEND, WICHTIG, NORMAL = Value
  }

}

class Verkaeufer(val vName: String, val vVorname: String, val vId: String)  {
  private var lohnProZeiteinheit: Int = 3
  var sumVerkaeufe: Double = 0.0


  def verkaufeDienstleitung(kunde: Kunde, prio: Verkaeufer.Prioritaet.Prioritaet, fehlerbeschreibung: String, m: Mechaniker) {
    val startzeit: Long = compat.Platform.currentTime
    this synchronized {
        this.wait((Math.random * 100).asInstanceOf[Int])
    }
    val endzeit: Long = compat.Platform.currentTime
    val dauer: Int = (endzeit - startzeit).toInt
    val lohn: Int = dauer * lohnProZeiteinheit
    sumVerkaeufe += lohn
    kunde.addBezahlt(lohn)
    m.bearbeiteJob(kunde, prio, fehlerbeschreibung)
  }

}
